nixos/modules/programs/java.nix

```diff
@@ -40,12 +38,38 @@ in
         type = types.package;
       };
 
+      binfmt = mkOption {
+        type = types.bool;
+        default = false;
+        description = lib.mdDoc "Register Java program execution with binfmt";
+      };
+
     };
 
   };
 
   config = mkIf cfg.enable {
 
+    boot.binfmt.registrations = mkIf cfg.binfmt {
+      java-class = {
+        recognitionType = "extension";
+        magicOrExtension = "class";
+        interpreter = pkgs.writeShellScript "java-class-wrapper" ''
+          file=''${1%%.class}
+          file=''${file/.\//}
+          java $file
```

Comment:
Why does this one not also source the setup-hook and use `$JAVA_PATH` and `"$@"`?

Reply:
I didn't know there's a setup-hook. I used the [Arch Linux Wiki](https://wiki.archlinux.org/title/Binfmt_misc_for_Java) to help do this.

Reply:
I mean like the jar one:

https://github.com/NixOS/nixpkgs/blob/ee1e3982c4328bbc8e07ad1583088c47ffe83578/nixos/modules/programs/java.nix#L66-L69

Reply:
I'll add it, I think I forgot about it or something.